src/main/java/it/pagopa/pn/deliverypush/externalclient/pnclient/safestorage/PnSafeStorageClientImpl.java

```diff
@@ -0,0 +1,92 @@
+    private final FileUploadApi fileUploadApi;
+    private final PnDeliveryPushConfigs cfg;
+    private final RestTemplate restTemplate;
+
+    public PnSafeStorageClientImpl(@Qualifier("withTracing") RestTemplate restTemplate, PnDeliveryPushConfigs cfg) {
+        it.pagopa.pn.delivery.generated.openapi.clients.safestorage.ApiClient newApiClient = new it.pagopa.pn.delivery.generated.openapi.clients.safestorage.ApiClient( restTemplate );
+        newApiClient.setBasePath( cfg.getSafeStorageBaseUrl() );
+
+        this.fileDownloadApi = new FileDownloadApi( newApiClient );
+        this.fileUploadApi =new FileUploadApi( newApiClient );
+        this.restTemplate = restTemplate;
+        this.cfg = cfg;
+    }
+
+
+    public FileDownloadResponse getFile(String fileKey, Boolean metadataOnly) {
+        log.debug("Start call getFile - fileKey {} metadataOnly {}", fileKey, metadataOnly);
+
+        return fileDownloadApi.getFile( fileKey, this.cfg.getSafeStorageCxId(), metadataOnly );
+    }
+
+    @Override
+    public FileCreationResponse createAndUploadContent(FileCreationWithContentRequest fileCreationRequest) {
+            log.debug("Start call createAndUploadFile - documentType {} filesize:{}", fileCreationRequest.getDocumentType(), fileCreationRequest.getContent().length);
+
+            String sha256 = DigestUtils.sha256Hex(fileCreationRequest.getContent());
+
+            FileCreationResponse fileCreationResponse = fileUploadApi.createFile( this.cfg.getSafeStorageCxId(),  "SHA-256", sha256,  fileCreationRequest );
+
+            log.debug("createAndUploadContent create file preloaded sha256:{}", sha256);
+
+            this.uploadContent(fileCreationRequest, fileCreationResponse, sha256);
+
+            log.info("createAndUploadContent file uploaded successfully key:{}", fileCreationResponse.getKey());
+
+            return fileCreationResponse;
+    }
+
+    private void uploadContent(FileCreationWithContentRequest fileCreationRequest, FileCreationResponse fileCreationResponse, String sha256){
+
+        try {
+            MultiValueMap<String, String> headers = new LinkedMultiValueMap<>();
+            headers.add("Content-type", fileCreationRequest.getContentType());
+            headers.add("x-amz-checksum-sha256", sha256);
+            headers.add("x-amz-meta-secret", fileCreationResponse.getSecret());
+
+            HttpEntity<Resource> req = new HttpEntity<>(new ByteArrayResource(fileCreationRequest.getContent()), headers);
+            ResponseEntity<Object> res = restTemplate.exchange( URI.create(fileCreationResponse.getUploadUrl()),
+                    fileCreationResponse.getUploadMethod()== FileCreationResponse.UploadMethodEnum.POST? HttpMethod.POST:HttpMethod.PUT, req, Object.class);
+            if (res.getStatusCodeValue() != org.springframework.http.HttpStatus.OK.value())
+            {
+                throw new PnInternalException("File upload failed");
+            }
+        } catch (PnInternalException ee)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An exception is caught and re-thrown within the catch clause without any other operations being performed on the exception, making the catch clause unnecessary.

**Fix**: Perform additional operations on the exception that has been caught. You can wrap a standard exception into a custom exception to add additional information or special handling. When you do that, make sure to set the original exception as the cause. Otherwise, you will lose the stack trace and message of the original exception, which will make it difficult to analyze the event that caused the exception.  
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)